clang/runtime/dpct-rt/include/dpct/codepin/codepin.hpp

```diff
@@ -1,83 +1,212 @@
-//==---- codepin.hpp -------------------------*- C++ -*---------------------==//
+//==---- codepin.hpp -------------------------*- C++ -*------------------==//
 //
 // Copyright (C) Intel Corporation
 // SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
 // See https://llvm.org/LICENSE.txt for license information.
 //
-//===----------------------------------------------------------------------===//
+//===-------------------------------------------------------------------===//
 #ifndef __DPCT_CODEPIN_HPP__
 #define __DPCT_CODEPIN_HPP__
 
-#include "detail/json.hpp"
-#include "detail/schema.hpp"
+#include "serialization/serial_basic.hpp"
+#include <fstream>
+#include <iomanip>
+#include <map>
 #include <memory>
+#include <numeric>
+#include <sstream>
+#include <string>
+#include <type_traits>
+#include <unordered_set>
+#include <utility>
+#include <vector>
+namespace dpct {
+namespace experimental {
+
+namespace detail {
+inline static std::unordered_set<void *> ptr_unique;
+inline static std::map<std::string, int> api_index;
+inline static std::string dump_file = "dump_log.json";
```

Comment:
Please add comment for the dump_file, including what kinds of content it has.

Reply:
Will do in another pr, track in https://jira.devtools.intel.com/browse/CTST-6392